exec-workitem/src/main/java/org/jbpm/process/workitem/exec/ExecWorkItemHandler.java

```diff
@@ -82,11 +96,24 @@ public void executeWorkItem(WorkItem workItem,
 
             parsedCommandStr = commandLine.toString();
 
+            ExecuteWatchdog watchdog = new ExecuteWatchdog(java.time.Duration.ofSeconds(defaultTimeoutInSeconds).toMillis());
             DefaultExecutor executor = new DefaultExecutor();
             ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
             PumpStreamHandler streamHandler = new PumpStreamHandler(outputStream);
             executor.setStreamHandler(streamHandler);
-            executor.execute(commandLine);
+		    executor.setWatchdog(watchdog);
+            try {
+				  executor.execute(commandLine);
+			    } catch (IOException e) {
+			        if (watchdog.killedProcess()) {
+			        	logger.debug("A timeout occured after "+java.time.Duration.ofSeconds(defaultTimeoutInSeconds).toMillis()+"ms while executing a command "+parsedCommandStr.replace(",", ""));
+						results.put(RESULT, outputStream.toString());
```

Comment:
If there was a timeout error / exception then I assume outputstream is empty ? wonder if you should return a default value or none at all here.

Reply:
outputstream do have a partial response for ping command. But sending partial response is not good idea. I have added failure message in result object in case of timeout and also added logger to log error.